mux.go

```diff
@@ -344,6 +354,11 @@ type RouteMatch struct {
 	Route   *Route
 	Handler http.Handler
 	Vars    map[string]string
+
+	// MatchErr is set to appropriate matching error
+	// It is set to ErrMethodMismatch if there is a mismatch in
+	// the request method and route method
+	MatchErr error
```

Comment:
This doesn't feel right: if you provide an `error` here you can't are extremely limited in how you can customize 405 responses.

Should we not offer a `MethodNotAllowedHandler` field on the `Router` akin to `NotFoundHandler` that a package user can customize? See https://godoc.org/github.com/gorilla/mux#Router

Reply:
@elithrar @kisielk Can we look into merging this PR if no other change is required? Or do you guys want to add `methodNotAllowedHandler` in this as well?

Reply:
@maknahar If you can add that as well, we can merge - that way it's feature-complete :)

Reply:
Sure.